File: Frontend/src/app/models/product.ts
import { Category } from './category';

export class Product {
  id: number;
  name: string;
  description: string;
  price: number;
  image: string;
  category: Category;

  constructor(res: Product) {
    this.id = res.id;
    this.name = res.name;
    this.description = res.description;
    this.price = res.price;
    this.image = res.image;
    this.category = res.category;
  }
}

File: Backend/src/main/java/com/example/backend/model/Product.java
package com.example.backend.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.*;
import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity(name = "products")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private Double price;
    @Lob
    private String image;

    @OneToOne
    private Category category;

    @ManyToMany(mappedBy = "products", cascade=CascadeType.PERSIST)
    //@JsonBackReference
    @JsonIgnore
    private List<Order> orders = new ArrayList<>();
}
